### PR TITLE
Amir yunas

### PR DESCRIPTION
**Pull request by ayunas**
@jheinz1999 

finally finished the TODO App.  Take a look and please provide any feedback you have about the code and styling.  Thanks alot!

### CODE REVIEW

**Comment by ayunas** [comment]
http://todoapp.lambdaschool.me
Here's the URL for the live App, in AWS.  thanks.

File: src/App.js [hunk 1]
```diff
@@ -1,16 +1,78 @@
 import React from 'react';
+import './styles/app.scss';
+import Header from './components/TodoComponents/Header';
```
Comment (jheinz1999) [inline src/App.js:3]: Great job using different components!

File: src/App.js [hunk 2]
```diff
@@ -1,16 +1,78 @@
 import React from 'react';
+import './styles/app.scss';
```
Comment (jheinz1999) [inline src/App.js:2]: Awesome that you're using scss!

File: src/App.js [hunk 3]
```diff
@@ -1,16 +1,78 @@
...
-  render() {
-    return (
-      <div>
-        <h2>Welcome to your Todo App!</h2>
-      </div>
-    );
-  }
+    constructor(props) {
+        super(props);
+        this.deleteTodos = this.deleteTodos.bind(this);
```
Comment (jheinz1999) [inline src/App.js:13]: I always use arrow functions so I don't have to bind the this keyword

File: src/App.js [hunk 4]
```diff
@@ -1,16 +1,78 @@
...
+        this.decideTodo = this.decideTodo.bind(this);
+        this.handleAddTodo = this.handleAddTodo.bind(this);
+        this.deleteTodo = this.deleteTodo.bind(this);
+        this.state = {
+            todos: []
+        }
+    }
+
+    deleteTodos() {
+        this.setState( () => ( { todos: [] } ) )
```
Comment (jheinz1999) [inline src/App.js:23]: It's cool that you used setState with a function, as we haven't covered that yet! You only really do this if you're updating data. The function can take in the previous state as a parameter

File: src/App.js [hunk 5]
```diff
@@ -1,16 +1,78 @@
...
+        this.setState( prevState => {
+            return {
+                todos : prevState.todos.filter( todo => todoToRemove !== todo )
+            }
+        }
+        )
+    }
+
+
+    decideTodo() {
```
Comment (jheinz1999) [inline src/App.js:37]: I love this function!

File: src/components/TodoComponents/VisibilityToggle.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        })
+    }
+
+    render() {
+        
+        return (
+            <div>
+                <h1>Visibility Toggle</h1>
+                <button onClick={this.handleToggleVisibility}>
+                {this.state.visibility ? 'hide details' : 'show details'}
```
Comment (jheinz1999) [inline src/components/TodoComponents/VisibilityToggle.js:25]: This is awesome!

**Review by jheinz1999 — APPROVED** [review]
Love what you did with this project! Looks really nice, and the code is excellent!

**Pull request merged**